File: Core-backend-layer/Programming-fundamentals/Java/OOP/src/Inheritance/ClothingProduct.java
class ClothingProduct extends Product {
    // Additional field
    private String size;
    private String material;

    public ClothingProduct(int productId, String name, double price, String manufacturer, String size, String material) {
        super(productId, name, price, manufacturer); // Call the superclass constructor
        this.size = size;
        this.material = material;
    }

    // Getter and Setter for the additional field
    public String getSize() {
        return size;
    }

    public void setSize(String size) {
        this.size = size;
    }

    public String getMaterial() {
        return material;
    }

    public void setMaterial(String material) {
        this.material = material;
    }

    // Override the displayInfo method to include additional information
    @Override
    public void displayInfo() {
        super.displayInfo(); // Call the superclass method
        System.out.println("Size: " + size + ", Material: " + material);
    }
}

File: Core-backend-layer/Programming-fundamentals/Python/OOP/src/inheritance/clothing_product.py
from product import Product

class ClothingProduct(Product):
    def __init__(self, productId, name, price, manufacturer, size, material):
        super().__init__(productId, name, price, manufacturer)
        self._size = size
        self._material = material

    # Getter and Setter for the additional field
    def getSize(self):
        return self._size

    def setSize(self, size):
        self._size = size

    def getMaterial(self):
        return self._material

    def setMaterial(self, material):
        self._material = material

    # Override the displayInfo method to include additional information
    def displayInfo(self):
        super().displayInfo()
        print(f"Size: {self._size}, Material: {self._material}")